anta/tests/routing/bgp.py

```diff
@@ -171,19 +83,43 @@ def _add_bgp_routes_failure(
     return failure_routes
 
 
-class VerifyBGPPeerCount(AntaTest):
-    """Verifies the count of BGP peers for a given address family.
+def _check_bgp_neighbor_capability(capability_status: dict[str, bool]) -> bool:
+    """Check if a BGP neighbor capability is advertised, received, and enabled.
 
-    It supports multiple types of Address Families Identifiers (AFI) and Subsequent Address Family Identifiers (SAFI).
+    Parameters
+    ----------
+    capability_status
+        A dictionary containing the capability status.
+
+    Returns
+    -------
+    bool
+        True if the capability is advertised, received, and enabled, False otherwise.
+
+    Example
+    -------
+    >>> _check_bgp_neighbor_capability({"advertised": True, "received": True, "enabled": True})
+    True
+    """
+    return all(capability_status.get(state, False) for state in ("advertised", "received", "enabled"))
+
+
+class VerifyBGPPeerCount(AntaTest):
+    """Verifies the count of established BGP peers with negotiated AFI/SAFI for given address families.
 
-    For SR-TE SAFI, the EOS command supports sr-te first then ipv4/ipv6 (AFI) which is handled automatically in this test.
+    This test performs the following checks for each specified address family:
 
-    Please refer to the Input class attributes below for details.
+      1. Confirms that the specified VRF is configured.
+      2. Counts the number of peers that are:
```

Comment:
Hmm this is breaking right? before we were "just" counting

Reply:
Right. As discussed, let's add a flag like you mentionned in your requested changes.

Reply:
Updated the test.

Reply:
We should update this to specify that if `check_peer_state: true` we also check the `Established` state + AFI/SAFI